lib/pengine/status.c

```diff
@@ -59,13 +59,25 @@ check_for_deprecated_rules(pcmk_scheduler_t *scheduler)
 gboolean
 cluster_status(pcmk_scheduler_t * scheduler)
 {
+    // @TODO Deprecate, replacing with a safer public alternative if necessary
     const char *new_version = NULL;
     xmlNode *section = NULL;
 
     if ((scheduler == NULL) || (scheduler->input == NULL)) {
         return FALSE;
     }
 
+    if (pcmk_is_set(scheduler->flags, pcmk__sched_have_status)) {
+        /* cluster_status() has already been called since the last time the
+         * scheduler was reset. Unpacking the input CIB again would cause
+         * duplication within the scheduler object's data structures.
+         *
+         * The correct return code here is not obvious. Nothing internal checks
+         * the code, however.
+         */
+        return TRUE;
+    }
+
```

Comment:
You might find what I started in #3735 interesting here.  I'd like to return to making progress on that, but I got bogged down with handling all the problems in our scheduler regression tests.

Reply:
I do, and in fact I mentioned that PR in a comment on this one:
https://github.com/ClusterLabs/pacemaker/pull/3810#issuecomment-2614151777